packages/SwingSet/src/liveslots/liveslots.js

```diff
@@ -438,6 +436,46 @@ function build(
     return Remotable(iface);
   }
 
+  let idCounters;
+  let idCountersAreDirty = false;
+
+  function initializeIDCounters() {
+    if (!idCounters) {
+      const raw = syscall.vatstoreGet('idCounters');
+      if (raw) {
+        idCounters = JSON.parse(raw);
+      } else {
+        idCounters = {};
```

Comment:
I think this case needs a `idCountersAreDirty=` so we'll do a write during `startVat`, instead of waiting until the first delivery that allocates something

Reply:
The base state always the same (`{}`) so we don't have to write it until there's an actual change.